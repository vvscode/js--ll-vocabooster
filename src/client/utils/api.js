const getFormdata = data => {
  const formData = new FormData();
  Object.keys(data).forEach(key => formData.append(key, data[key]));
  return formData;
};

export const getWords = text =>
  fetch('/api/vocagrabber', {
    method: 'POST',
    body: getFormdata({ text }),
  })
    .then(r => r.json())
    .then(({ data: list }) =>
      list.map((item, index) => ({ ...item, id: index })),
    );

export const addWordsToLingualeoVocabulary = words => Promise.resolve();
