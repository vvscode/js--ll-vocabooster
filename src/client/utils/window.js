export const { document } = window;

export const setTitle = title => {
  window.document.title = title;
};

export default window;
