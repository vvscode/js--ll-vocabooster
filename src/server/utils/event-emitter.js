import { EventEmitter } from 'events';

const customEventEmitter = new EventEmitter();

export default customEventEmitter;
