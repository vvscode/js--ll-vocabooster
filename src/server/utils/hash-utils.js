import hash from 'object-hash';

const hashAny = (value, keysToExclude = []) => {
  if (!value) return {};

  return {
    val: value,
    key: hash(value, {
      algorithm: 'md5',
      excludeKeys: key => keysToExclude.includes(key),
    }),
  };
};

const hashArray = arr => {
  const newArray = [];

  arr.forEach(item => {
    newArray.push(hashAny(item));
  });

  return newArray;
};

export { hashArray, hashAny };
