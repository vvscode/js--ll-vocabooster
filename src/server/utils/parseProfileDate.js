import dateformat from 'dateformat';

export default function parseProfileDate(dateString) {
  try {
    return dateformat(dateString, 'isoDate');
  } catch (error) {
    console.error(error);
    return null;
  }
}
