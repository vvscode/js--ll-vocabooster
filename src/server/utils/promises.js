export const serialProcess = (list, cb) => {
  let promise = Promise.resolve();
  const listCopy = [...list];
  listCopy.forEach(item => {
    promise = promise.then(() => cb(item));
  });
  return promise;
};

export const sleep = delay =>
  new Promise(resolve => setTimeout(resolve, delay));
