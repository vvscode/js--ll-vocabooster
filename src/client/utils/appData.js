import window from './window';

const data = (window && window.App) || {};

export const isLoggedIn = () =>
  !!data.user && Object.keys(data.user).length > 0;

export default data;
