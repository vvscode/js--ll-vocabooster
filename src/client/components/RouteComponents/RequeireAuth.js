import React from 'react';
import { Redirect } from 'react-router';
import { isLoggedIn } from 'client/utils/appData';

export default (component, redirectTo = '/') =>
  isLoggedIn() ? component : () => <Redirect to={redirectTo} />;
