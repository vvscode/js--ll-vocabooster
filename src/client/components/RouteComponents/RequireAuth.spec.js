import React from 'react';
import { strictEqual } from 'assert';

import * as appData from 'client/utils/appData';
import RequeireAuth from './RequeireAuth';

jest.mock('client/utils/appData');

describe('RequeireAuth', () => {
  let DummyComponent;
  beforeEach(() => {
    DummyComponent = () => <div>Dummy</div>;
  });

  it('renders redirect if user non-authenticated', () => {
    appData.isLoggedIn.mockReturnValue(false);
    expect(RequeireAuth(DummyComponent)()).toMatchSnapshot();
  });

  it('renders redirect to certain url if user non-authenticated and url passed', () => {
    appData.isLoggedIn.mockReturnValue(false);
    expect(
      RequeireAuth(DummyComponent, '/some-magic-path')(),
    ).toMatchSnapshot();
  });

  it('returns component itself if user is authenticated', () => {
    appData.isLoggedIn.mockReturnValue(true);
    strictEqual(RequeireAuth(DummyComponent), DummyComponent);
  });
});
