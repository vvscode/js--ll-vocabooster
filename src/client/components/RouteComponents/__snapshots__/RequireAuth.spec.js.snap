// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequeireAuth renders redirect if user non-authenticated 1`] = `
<Redirect
  push={false}
  to="/"
/>
`;

exports[`RequeireAuth renders redirect to certain url if user non-authenticated and url passed 1`] = `
<Redirect
  push={false}
  to="/some-magic-path"
/>
`;
